### ISSUE TITLE
Astropy equivalencies not fully supported

### ISSUE THREAD

**Issue by adrn** [GalacticDynamics/unxt#234]
We have this doctest example:
```python
x = Quantity([1, 2, 3], "Kelvin")
with u.add_enabled_equivalencies(u.temperature()):
    y = x.to(u.deg_C)
```
which works, but this one doesn't work:
```python
x = Quantity([1, 2, 3], "radian")
with u.add_enabled_equivalencies(u.dimensionless_angles()):
    y = x.to(u.one)
```

I think this is because the physical type (dimension) changes in the latter case and not the first. See https://github.com/GalacticDynamics/unxt/blob/main/src/unxt/_src/quantity/base_parametric.py#L43

**Comment by adrn** [comment]
Actually I think the main issue is with [this line](https://github.com/GalacticDynamics/unxt/blob/main/src/unxt/_interop/unxt_interop_astropy/quantity.py#L315) - I don't think we want to do a `replace` here (that line just replaces the unit and value of a quantity), because conversions with equivalencies can change the physical type (dimensions) of a quantity (such as with `dimensionless_angles()`).

**Comment by adrn** [comment]
Right. Though do we expect other arguments? If we don't mind type coercion, one option would be to swap the `replace` for:
```python
return Quantity(_apy7_unit_to(x.unit, unit, x.value), unit)
```

**Comment by nstarman** [comment]
We can support all fields by using the `dataclassish.field_items()`
Preventing unnecessary type coercion is a bit more fiddly. I think we can do this by checking the dimensions of the output unit against the input units and only doing type coercion if they are different.
